### PR TITLE
fix(button): set default button heights

### LINKED ISSUE #1691: [Carbon X] Implicit height on Button can sometimes render as zero height
## Detailed description

Button height for Carbon X version is implicit:
https://github.com/IBM/carbon-components/blob/master/src/components/button/_mixins.scss#L84

This usually doesn't cause issues for most buttons, since they typically have content within.

However, the `ButtonSkeleton` from `carbon-component-react` renders with a zero height due to this implicit `height: auto` setting.  

## Suggested Fix

Adding a pixel value for height resolves in the Carbon X button base mixin resolves this issue.

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1691

This PR sets default button heights to avoid collapsing when button content is not rendered

### CODE REVIEW

File: src/components/button/_mixins.scss [hunk 1]
```diff
@@ -81,7 +81,7 @@
   flex-shrink: 0;
   font-size: $button-font-size;
   font-weight: $button-font-weight;
-  height: auto;
+  height: rem(48px);
```
Comment (jendowns) [inline src/components/button/_mixins.scss]: IMO this should be `rem($button-height)` -- see: https://github.com/IBM/carbon-components/blob/e4da4a57542a42f7c6e0a90e91aac9b7e0fcaf02/src/globals/scss/_theme-tokens.scss#L140

**Review by jendowns — CHANGES_REQUESTED** [review]

File: src/components/button/_button.scss [hunk 2]
```diff
@@ -305,7 +305,7 @@
 
   .#{$prefix}--btn--sm {
     @include letter-spacing;
-    height: auto;
+    height: rem(32px);
```
Comment (jendowns) [inline src/components/button/_button.scss:308]: Just pointing out that this is a bit different than how height is applied to small buttons in the non-experimental theme:
https://github.com/IBM/carbon-components/blob/98bae993a38f43dc45e6a23b5302f882926a0ab1/src/components/button/_button.scss#L163

Comment (emyarod) [reply src/components/button/_button.scss:308]: the output is identical (2rem) as our base font size is 16px, unless I am mistaken. Please correct me if that's not the case

Comment (jendowns) [reply src/components/button/_button.scss:308]: Yup, it's the same. Just pointing it out for consistency 👍

Comment (emyarod) [reply src/components/button/_button.scss:308]: yeah I figured I would use the `rem` conversion function to be consistent with the rest of the experimental styles

**Review by asudoh — COMMENTED** [review]
Thank you for jumping in @emyarod! One question - I thought the original issue pointed out a wrong token of `$button-height` in experimental mode. Did you find otherwise?

**Comment by emyarod** [comment]
you're right @asudoh I overlooked that var and hardcoded its value instead. Should be fixed now

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Review by tw15egan — APPROVED** [review]

**Pull request merged**